nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSStateFlattener.java

```diff
@@ -37,17 +38,31 @@
 public class DBSStateFlattener {
 
     public static Map<String, Serializable> flatten(State state) {
+        return flatten(state, Collections.emptyMap());
+    }
+
+    /**
+     * Flatten with optional property mappings
+     * @param state State
+     * @param keyMappings A map of key/values to replace
+     * @return Flattened result
+     * @since 9.3
+     */
+    public static Map<String, Serializable> flatten(State state, Map<String, String> keyMappings) {
```

Comment:
Please make all methods non-`static` and pass the `keyMappings` in the constructor, to avoid propagating it in all recursive calls.

    protected static final Map<String, String> keyMappings;

    public DBSStateFlattener(Map<String, String> keyMappings) {
        this.keyMappings = keyMappings;
    }

This will require changes to the few callers of course: `new DBSStateFlattener(mappings).flatten(state)`